### DISCUSSION TITLE
Navigation Bar Color Override Option

### DISCUSSION THREAD

**Discussion by iotdetective** [requarks/wiki#7147] [Guides]
Wanted to share this CSS override for the site navigation bar color change, since it was a question I didn't see answered online elsewhere and everything I read says that it currently is not possible.  My workaround doesn't apply to the Wiki.js administration area, but does apply to the public facing sections of my Wiki.js site (minus the the tag explorer page for some reason).  When I couldn't find a solution for this modification I started exploring things with the inspect element tool in Google Chrome, I was able to come up with the following sample CSS Override which you can add in the Administration Area under: Themes > Code Injection > CSS Override and click apply:

![image](https://github.com/requarks/wiki/assets/88954886/8d1b268b-5f34-4600-b239-c4e6b92bdc7a)

Easy copy and paste code (just change to the desired color):
`.nav-header-inner .v-toolbar__content
{
  background-color: midnightblue;
}
.v-application .black
{
  background-color: midnightblue !important;
}`

I am using the Dark Theme, but I just tested it with the light theme and it works.

Proof of Concept:

![image](https://github.com/requarks/wiki/assets/88954886/5119af0e-6079-40a6-80ec-7b54b5d4fcfe)

**Comment by Tbwan** [comment]
awesome, I had been looking for this. Have you found a way to edit the icon colors and search bar? I'm interested in changing them to a light/white color scheme.

edit:
I figured it out, here's the changes I made:

```
.v-application .grey--text {
  color: #fff!important
}
.v-application .grey--text {
  caret-color: #fff!important
}
.v-input__slot {
  background: #fff!important
}
.v-toolbar__content .v-input,
.v-toolbar__content .v-input input,
.v-toolbar__content .v-input textarea {
  color: #333!important
}
.v-input__slot .v-icon {
  color: #707070!important
}
.v-text-field__slot .v-label {
  color: #707070!important
}
```

**Reply by iotdetective** [reply]
Nice work!  Sorry for the delayed response I missed your original comment, this week has been crazy.  Hopefully people keep digging into these custom CSS codes. I've been running Wiki.js for a year or so now, and there have been several great updates but the themes section has been "coming soon" for a while now.  Understanding the development takes time, maybe in the meantime Wiki.js could add some of these options to their documentation for those looking for a temporary fix until themes are fully implemented and customizable.

**Comment by iamlucas13** [comment]
> Je voulais partager cette substitution CSS pour le changement de couleur de la barre de navigation du site, car c'était une question à laquelle je n'avais pas vu de réponse en ligne ailleurs et tout ce que j'ai lu dit que ce n'est actuellement pas possible. Ma solution de contournement ne s'applique pas à la zone d'administration de Wiki.js, mais s'applique aux sections publiques de mon site Wiki.js (moins la page d'exploration des balises pour une raison quelconque). Lorsque je n'ai pas pu trouver de solution pour cette modification, j'ai commencé à explorer les choses avec l'outil d'inspection des éléments dans Google Chrome, j'ai pu trouver l'exemple de substitution CSS suivant que vous pouvez ajouter dans la zone d'administration sous : Thèmes > Injection de code > Substitution CSS et cliquez sur Appliquer :
> 
> ![image](https://private-user-images.githubusercontent.com/88954886/326308743-8d1b268b-5f34-4600-b239-c4e6b92bdc7a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY4NzMzNjAsIm5iZiI6MTczNjg3MzA2MCwicGF0aCI6Ii84ODk1NDg4Ni8zMjYzMDg3NDMtOGQxYjI2OGItNWYzNC00NjAwLWIyMzktYzRlNmI5MmJkYzdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTE0VDE2NDQyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5ZTcwYmI0Mzg5Y2Q1M2FiOGI0NDdlMjViY2RmYmVjNWFlMzIyODgyMDkwYTk3MGU5N2FlNmUwOWUwYTNjODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tEqD2aEfE3n74DlyRcRd1tOOvYjhPMszSeHoFIf6Fkc)
> 
> Copiez et collez facilement le code (il suffit de changer la couleur souhaitée) : `.nav-header-inner .v-toolbar__content { background-color: midnightblue; } .v-application .black { background-color: midnightblue !important; }`
> 
> J'utilise le thème sombre, mais je viens de le tester avec le thème clair et cela fonctionne.
> 
> Preuve de concept :
> 
> ![image](https://private-user-images.githubusercontent.com/88954886/326308716-5119af0e-6079-40a6-80ec-7b54b5d4fcfe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY4NzMzNjAsIm5iZiI6MTczNjg3MzA2MCwicGF0aCI6Ii84ODk1NDg4Ni8zMjYzMDg3MTYtNTExOWFmMGUtNjA3OS00MGE2LTgwZWMtN2I1NGI1ZDRmY2ZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTE0VDE2NDQyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0MTJmMGEyNzIwOGY5N2ZmZTliN2YxMDc5YTE3MmM5NjcxNDA1NDg4MTBmNTJhZGM4MmE1NWQzYjMxYjNiMTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zDB-k_88MbvCALtifZGbn6jSbT6H2c5G1K-fzga7WKI)

Hello, for me working but I have two black line at the corner : 

![image](https://github.com/user-attachments/assets/99f268d2-08e6-499f-963f-368ba2368194)

Do you have an idea?

Best

**Reply by RichardDavidKaavalan** [reply]
Remove the css override code and add this in under: Themes > Code Injection > Body HTML Injection
<style type="text/css"> .v-application .black { background-color: #074e7a !important; border-color: #074e7a!important } </style>

hope this helps!
